doc/basics.md

```diff
@@ -57,9 +57,20 @@ Once you have an element, you can navigate it with idiomatic C++ iterators, oper
 
 * **Extracting Values:** You can cast a JSON element to a native type: `double(element)` or
   `double x = json_element`. This works for double, uint64_t, int64_t, bool,
-  dom::object and dom::array. You can also use is_*typename*()` to test if it is a
-  given type, and as_*typename*() to do the cast and return an error code on failure instead of an
-  exception.
+  dom::object and dom::array. An exception is thrown if the cast is not possible. You can also use is<*typename*>() to test if it is a
+  given type, or use the `type()` method: e.g., `element.type() == dom::element_type::DOUBLE`. Instead of casting, you can use get<*typename*>() to get the value: casts and get<*typename*>() can be used interchangeably. You can use a variant usage of get<*typename*>() with error codes to avoid exceptions: e.g.,  
+  ```c++
+    simdjson::error_code error;
+    double value;
+    result.get<double>().tie(value, error); // no exception, ever
+    /**
+     * If you do not have to support libc++, you can use the simpler syntax:
+     * auto [value, error] = result.get<double>();
+     */
+    if(error) {
+      // error case
+    }
+    ```
```

Comment:
The code and trailing \`\`\` seems to be misaligned from the \`\`\`c++, I don't know if that's going to render right on all markdown viewers ...

Reply:
Fixed